metadata-ingestion/src/datahub/ingestion/source/unity/source.py

```diff
@@ -269,10 +232,11 @@ def build_service_principal_map(self) -> None:
     def process_metastores(self) -> Iterable[MetadataWorkUnit]:
         metastores: Dict[str, Metastore] = {}
         assigned_metastore = self.unity_catalog_api_proxy.assigned_metastore()
-        if assigned_metastore:
-            metastores[assigned_metastore.metastore_id] = assigned_metastore
-        for metastore in self.unity_catalog_api_proxy.metastores():
-            metastores[metastore.metastore_id] = metastore
+        metastores[assigned_metastore.metastore_id] = assigned_metastore
+
+        if self.config.ingest_all_metastores:
+            for metastore in self.unity_catalog_api_proxy.metastores():
+                metastores[metastore.metastore_id] = metastore
```

Comment:
By default, now only ingest the "current" metastore, so we don't need admin privs. I haven't actually tested with multiple metastores, but have tested with `ingest_all_metastores` on and off.